### ISSUE TITLE
Make the whole qubes-builder deterministic

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#816]
**Reported by joanna on 11 Apr 2014 09:09 UTC**
... to allow easy, independent comparison of rpms/ISOs build from the same sources on different machines and by different people.

In the future: allow rpm/iso signatures by ""M out of N" signers.

Migrated-From: https://wiki.qubes-os.org/ticket/816

**Comment by marmarek** [comment]
Useful link for Fedora work on reproducible builds:
https://github.com/kholia/ReproducibleBuilds
http://securityblog.redhat.com/2013/09/18/reproducible-builds-for-fedora/

**Comment by andrewdavidwong** [comment]
> In the future: allow rpm/iso signatures by ""M out of N" signers.

Since this is actually a separate issue, I've created #2535.

**Comment by marmarek** [comment]
Status:
 - Debian packages for R4.0 build twice in the same environment are reproducible (at least we don't have timestamp issues)
 - Fedora packages for R4.0 dom0 are not (too old rpm), R4.1 should be fine
 - Fedora packages for fc30 VM mostly are reproducible (tested local build vs travis) - with exception to very few packages like core-agent-linux (see https://github.com/QubesOS/qubes-issues/issues/4625)

Besides packages itself, some tooling is missing to reproduce environment (frozen set of dependencies). We do have buildinfo files, but not an automated way to prepare build environment based on it.

cc @iprid23

**Comment by garlicgambit** [comment]
@marmarek what is the current status of reproducible builds? Is the Qubes install iso deterministically reproducible?

**Comment by marmarek** [comment]
No, it isn't. Not much changed from the comment above, but we do have scheduled some work on this later this year.

**Comment by garlicgambit** [comment]
What is the current status of reproducible builds for the Qubes install iso?
